acl/src/main/java/org/springframework/security/acls/jdbc/JdbcAclService.java

```diff
@@ -88,7 +88,7 @@ public JdbcAclService(DataSource dataSource, LookupStrategy lookupStrategy) {
 	// ========================================================================================================
 
 	public List<ObjectIdentity> findChildren(ObjectIdentity parentIdentity) {
-		Object[] args = { parentIdentity.getIdentifier(), parentIdentity.getType() };
+		Object[] args = { parentIdentity.getIdentifier().toString(), parentIdentity.getType() };
```

Comment:
Can you please account for the fact that this `getIdentifier()` might be `null`? Initially I thought this would be best with `String.valueOf`, but that converts `null` to `"null"` which is not ideal. Instead I think we should do a `null` check and if null use `null`.

NOTE: Rather than commenting in each spot, can you please apply this change throughout